cranelift/codegen/meta/src/shared/instructions.rs

```diff
@@ -1314,6 +1314,32 @@ pub(crate) fn define(
         .other_side_effects(true),
     );
 
+    ig.push(Inst::new(
+        "get_frame_pointer",
+        r#"
+        Get the address in the frame pointer register.
+
+        Usage of this instruction requires setting `preserve_frame_pointers` to `true`.
+        "#,
+        &formats.nullary
+    ).operands_out(vec![addr]));
+
+    ig.push(Inst::new(
+        "get_stack_pointer",
+        r#"
+        Get the address in the stack pointer register.
+        "#,
+        &formats.nullary
+    ).operands_out(vec![addr]));
+
+    ig.push(Inst::new(
+        "get_return_pointer",
```

Comment:
a tiny bikeshed-color nit, but I think usually this intrinsic refers to the "return address" rather than "return pointer" (e.g., `__builtin_return_address()` in GCC/Clang); perhaps we could call it that?